### PR TITLE
Nethui Tickets credential

### PR DESCRIPTION
**Pull request by rafaelbernard**

### CODE REVIEW

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+				<nl></nl>
+			</Name>
+			<Description>
+				<en>Event</en>
+				<nl></nl>
+			</Description>
+		</Attribute>
+		<Attribute id="eventDate">
+			<Name>
+				<en>Event date</en>
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: What if the event is across many dates?

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+			<Name>
+				<en>Ticket type</en>
+				<nl></nl>
+			</Name>
+			<Description>
+				<en>Ticket type</en>
+				<nl></nl>
+			</Description>
+		</Attribute>
+		<Attribute id="fullName">
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Do we need a name on the ticket if we used a name from our Attributes to get the ticket issued?

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+				<nl></nl>
+			</Name>
+			<Description>
+				<en>Full name</en>
+				<nl></nl>
+			</Description>
+		</Attribute>
+		<Attribute id="email">
+			<Name>
+				<en>E-mail</en>
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Do we need an email if we used an email attribute already on the phone?

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+				<nl></nl>
+			</Name>
+			<Description>
+				<en>E-mail</en>
+				<nl></nl>
+			</Description>
+		</Attribute>
+		<Attribute id="organisationName" required="false">
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Do we need an org name for a ticket?

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+				<nl></nl>
+			</Name>
+			<Description>
+				<en>Organisation name</en>
+				<nl></nl>
+			</Description>
+		</Attribute>
+		<Attribute id="membershipNumber" required="false">
+			<Name>
+				<en>INZ membership number</en>
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: We already used our membership attribute on the phone already to get the ticket, do we need to store it again?

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+			<Name>
+				<en>Purchase date</en>
+				<nl>Purchase date</nl>
+			</Name>
+			<Description>
+				<en>Date of purchase</en>
+				<nl>Date of purchase</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="price">
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Probably dont need the price, as it's an access pass - the type is likely the more important attribute.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 7]
```diff
@@ -0,0 +1,123 @@
+<IssueSpecification version="4">
+	<Name>
+		<en>Demo INZ Nethui Ticket</en>
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Let's call it a Nethui access pass

Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Because we're going to use it for access to many things rather than a ticket to 1 thing

**Review by camfindlay — CHANGES_REQUESTED** [review]
I think we can make the access pass scheme simpler and minimise stored data.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 1 above)
Comment (ehsan-fj) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Could be event start date?

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 2 above)
Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: NetHui ticket website enables you to buy a ticket for someone else. I know this is not a blocker for anything, but, as the form requires it, I think that could make sense to store the name that was informed at the form, that could be the same or different from the full name of the buyer.

Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: I imagine that this is a more relaxed credential.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 1 above)
Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: I see no problem. As an example, I am storing `13 and 14/10/2020` for NetHui 2020:

image

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 3 above)
Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Same as for the name. I believe this is a more relaxed credential. The buyer could be different from the ticket user.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 4 above)
Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: The organisation name is asked on the website, but I can remove from the ticket. But could be used inside (or online). For instance: "May I see your ticket? We are providing X for attendees from Y organisation.".

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 6 above)
Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: I am storing the price right now just to "prove" the discount was applied when membership was used.
But the price is sometimes stored at tickets access passes.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 3 above)
Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: I dont think someone would store someone else's ticket on their digital identity wallet as such given the use case for these wallets is peoples personal attributes and digital identity related attributes. We have to be careful here not to be duplicating other ticket services like eventbrite etc.

Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: If you bought a ticket for someone else, I'd think that it either emails that other person the ticket or that person gets sent the QR code to load into their own IRMA wallet.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 2 above)
Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: See below comments

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 6 above)
Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Cool sounds good for the PoC

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 4 above)
Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: The event would have these details in their database anyway all they need to to match the disclosed ticket number and they can work out the other information for the attendee.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 3 above)
Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Exactly. But you would load your credentials, but inform someone else's name and email. When the QR code appears, another person can scan the QR.

But I can remove if this does not make sense now.

Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Let's keep this simple, InternetNZ member can prove they are a member to get the discount for their access pass. Their access pass can be issued to their IRMA wallet proving they are allowed to attend the series of online and in-person nethui events (the user could buy online, in-person or both as ticket types). Should we then also build a simple online access website and a kiosk webpage to allow people to "attend" the events to finish off a nice overall PoC flow?

Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Let's not worry about issuing other's tickets for this PoC, we can just fake that bit.

Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: OK. I removed fullName and email.

A page to "attend" is good! Will do that.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 1 above)
Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Do I keep event date info?

Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Perhaps turn this into a "Valid until" date

Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Maybe you can buy a 1 day pass, or a pass to all the events. A one day pass might need a valid from and until date? So example might be I dont want to go to the online event, but I want a 1 day pass to attend the in-person networking event on the 2nd day of Nethui.

Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: I think `eventDate` still is more clear.

* 1 day = `13/10/2020` or
* 1 day = `14/10/2020` (only second day)
* 2 days = `13 and 14/10/2020` or
* 2 days = `13, 14/10/2020`
* One week = `13-20/10/2020`

Or any other format that we define.

Comment (camfindlay) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Sure, happy to go with that given it's a PoC.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+		<en>Demo INZ Nethui Access Pass</en>
+		<nl>Demo INZ Nethui Access Pass</nl>
+	</Name>
+	<ShortName>
+		<en>Demo INZ Nethui Access Pass</en>
+		<nl></nl>
+	</ShortName>
+	<SchemeManager>inz-demo</SchemeManager>
+	<IssuerID>internetnz</IssuerID>
+	<CredentialID>nethuiTicket</CredentialID>
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: nethuiAccessPass

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+	<Description>
+		<en>INZ Nethui Ticket</en>
```
Comment (camfindlay) [inline inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Nethui Access Pass

**Review by camfindlay — CHANGES_REQUESTED** [review]
drop the ticket terminology otherwise the rest is all good.

File: inz-demo/internetnz/Issues/nethuiTicket/description.xml (hunk 8 above)
Comment (rafaelbernard) [reply inz-demo/internetnz/Issues/nethuiTicket/description.xml]: Fair enough. I changed here and at demo UI.

**Review by camfindlay — APPROVED** [review]

**Review by ehsan-fj — APPROVED** [review]

**Pull request merged**